test/infrastructure/docker/controllers/dockercluster_controller.go

```diff
@@ -123,6 +127,7 @@ func reconcileNormal(ctx context.Context, dockerCluster *infrav1.DockerCluster,
 	// Set APIEndpoints with the load balancer IP so the Cluster API Cluster Controller can pull it
 	lbip4, err := externalLoadBalancer.IP(ctx)
 	if err != nil {
+		conditions.MarkFalse(dockerCluster, infrav1.LoadBalancerProvisionedCondition, infrav1.LoadBalancerProvisioningFailedReason, clusterv1.ConditionSeverityWarning, err.Error())
```

Comment:
It looks like this should actually be set prior to line 124, since that is where it's creating the load balancer. I'm wondering if this should possibly have a different failure reason other than provisioning failed.

Reply:
good catch!
WRT to adding a new reason, we can do this, but given that CAPD is not meant for production I prefer to keep it simple and only differentiate the message that documents why the provisioning is failed.